tests/acceptance/page_objects/base.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+
+
+class BasePage(object):
+    """Base class for PageObjects"""
+    page_name = 'base'
+
+    def __init__(self, browser):
+        self.browser = browser
+        try:
+            self.assert_correct_page()
+        except AssertionError:
+            raise Exception(
+                'This is not the %s page. Current url is %s.' %
+                (self.page_name, self.driver.current_url))
+
+    @property
+    def driver(self):
+        return self.browser.driver
+
+    def assert_correct_page(self):
+        pass
+
+
+class BaseElement(object):
+    def __init__(self, selector, element):
+        self.selector = selector
```

Comment:
If Element objects are passed the relevant DOM node what do you need the selector for? I don't see any of the implementations you have right now using the selector.

Reply:
You're right. A previous iteration of this code used the selector for things, but not anymore. Let me go ahead and remove that code